### DISCUSSION TITLE
Image filename with spaces

### DISCUSSION THREAD

**Discussion by timler** [facebook/docusaurus#10720] [Q&A] [answered]
I have a question about adding images with spaces in their name. I cannot get them to work in Docusaurus and am wondering if this is a bug or if I am doing something wrong.

I have two images under `static`:
* `test one.png`
* `test-two.png`

This markdown works, to insert an image:
```
![Testing](/img/test-two.png)
```

This markdown compiles, but does not insert the image
```
![Testing](/img/test one.png)
```

None of this markdown compiles - I see the error: `Image static/img/test%20one.png used in docs/developer/architecture.mdx not found`

```
![Testing](/img/test%20one.png)
![Testing](</img/test%20one.png>)
![Testing](</img/test one.png>)
```

I am using the latest version of Docusauraus (3.6.2) on Mac OS

Can anyone tell me if filenames with spaces are supported in Docusaurus?

Thanks!

**Comment by slorber** [comment]
Markdown links with spaces need to be wrapped in `<` `>` because what comes after the space could be an "extension" 

```
❌
![Testing](/img/test one.png)

✅
![Testing](</img/test one.png>)
```

Even though afaik it has never been standardized, it's possible that CommonMark someday allows something like `![alt](/img/myimg.png width=800px)`

**Reply by timler** [reply]
Thanks for the reply, unfortunately the angle brackets do not work for me. 

I tried all these and ALL of them gave the same error "Image static/img/test%20one.png used in docs/developer/architecture.mdx not found"

```
![Testing](/img/test%20one.png)
![Testing](</img/test%20one.png>)
![Testing](</img/test one.png>)
```

Have you got this working in your Docusaurus instance?

**Reply by slorber** [reply]
Hmmm yes there's something weird happening due to URI component encoding, it seems you need to name your image `test%20one.png` 😅 

It's a bug in our `tranformImage` remark plugin.

**Reply by slorber** [reply, marked as answer]
This bug will be fixed in the next release

PR https://github.com/facebook/docusaurus/pull/10723

**Comment by pranabdas** [comment]
You may use `img` tag instead:
```
<img src={require("/img/test one.png").default} alt="Testing"/>
```
